server/service/src/invoice/inventory_adjustment/add_new_stock_line/validate.rs

```diff
@@ -0,0 +1,45 @@
+use repository::StorageConnection;
+
+use crate::common_stock::check_stock_line_does_not_exist;
+
+use crate::stocktake_line::validate::{check_active_adjustment_reasons, check_reason_is_valid};
+use crate::validate::check_store_exists;
+
+use super::insert::{AddNewStockLine, AddNewStockLineError};
+
+pub fn validate(
+    connection: &StorageConnection,
+    store_id: &str,
+    input: &AddNewStockLine,
+) -> Result<(), AddNewStockLineError> {
+    use AddNewStockLineError::*;
+    if !check_store_exists(connection, store_id)? {
+        return Err(InvalidStore);
+    }
+
+    if !check_stock_line_does_not_exist(&input.stock_line_id, connection)? {
```

Comment:
I got a little bit confused here. Can we not generate stock line id ?

Reply:
Thanks for review! Yeah let’s chat on Thursday. 

Main reason for passing the stock line ID was to prevent getting duplicates if you send the same request many times… other option would be to instead send an invoice ID for the inventory adjustment, and generate stock line ID - this might be simpler, and it is what we've done for the other invoices... I just found myself calling the `insertStockLine` mutation and it felt like the right ID to include there would be a stock line ID?